Plugins/Skinning/src/SkinningSystem.hpp

```diff
@@ -17,9 +17,15 @@
 
 namespace SkinningPlugin
 {
+    class SkinningPluginC;
+}
+
+namespace SkinningPlugin
+{
 
     class SKIN_PLUGIN_API SkinningSystem :  public Ra::Engine::System
     {
+        friend class SkinningPlugin::SkinningPluginC;
```

Comment:
I still think it would be better to have an accessor function. I don't like the idea of the "plugin" class (which is related to UI) being used in the System class which is related to the Engine.